openedx/core/djangoapps/programs/utils.py

```diff
@@ -127,6 +127,43 @@ def engaged_programs(self):
 
         return programs
 
+    @cached_property
+    def failed_course_runs(self):
+        """
+        Determine which course runs have been failed by the user.
+
+        Returns:
+            list of strings, each representing a certificate course run key
+        """
+        course_run_certificates = certificate_api.get_certificates_for_user(self.user.username)
+        return [
+            unicode(certificate['course_key']) for certificate in course_run_certificates
+            if not certificate_api.is_passing_status(certificate['status'])
+        ]
+
+    def does_course_still_qualify_as_in_progress(self, now, enrolled_run_modes, course):
+        # Part 1: Check if any of the seats you are enrolled in qualify this course as in progress
+        enrolled_runs = [run for run in course['course_runs'] if run['is_enrolled']]
```

Comment:
Look at how `_is_course_in_progress()` determines whether the user is enrolled. The way you have things here will only work if an "extended program" is provided, which isn't a dependency we need or want.